[Shawn Tan Jing Kai] iP

shawnkai opened the pull request:
# Duke

> “Your mind is for having ideas, not holding them.” – David Allen

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ ***SUPER FAST to use***

All you need to do is,

1. download it from [here](https://github.com/shawnkai/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :+1:
And it is FREE!

Features:

 - [x] Managing tasks
 - [x] Managing deadlines
 - [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
    public static void main(String[] args) throws IOException {
        Storage storage = new Storage();
        TaskList tasks = storage.loadFile();
        Parser parser = new Parser(tasks, storage);
        UI ui = new UI();
        ui.welcomeUser();
        String input = ui.readInput();  // Read user input
        while (!input.equals("bye")) {
            parser.parse(input);
            input = ui.readInput(); // Read next user input
        }
        ui.sayBye();
    }
```

Commit:
Add Gradle support

shawnkai committed:
edited Duke.java. completed Level-1

shawnkai committed:
edited duke.java and completed level 2

shawnkai committed:
added task.java

shawnkai committed:
edited duke.java

shawnkai committed:
edited task.java and added 3 subclasses

shawnkai committed:
edited duke.java

shawnkai committed:
added DukeException.java and edited Duke.java

shawnkai committed:
edited Duke.java

shawnkai committed:
removed emoji from duke.java

shawnkai committed:
resolved A-TextUiTesting issue

shawnkai committed:
edits made to several files for level 7

shawnkai committed:
edited deadline.txt and event.txt so they can recognise dates

shawnkai committed:
Merge branch 'branch-Level-7'

shawnkai committed:
Merge branch 'branch-Level-8'

shawnkai committed:
Extracted code from Duke.java to form classes: Parser, Storage, TaskList, UI

shawnkai committed:
put all classes in one package duke

shawnkai committed:
Merge remote-tracking branch 'add-gradle-support/add-gradle-support'

shawnkai committed:
created test class for event and todo classes

shawnkai committed:
changes made to input.txt

shawnkai committed:
added data.txt file to gitignore

desmondyst commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+        this.by = LocalDate.parse(by);
+    }
+
+    @Override
```
Good that you remember to add Override annotations!

desmondyst commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,101 @@
+public class Parser {
+    private TaskList tasks;
+    private Storage storage;
+    String dummyString;
+    Task dummyTask;
+    int counter;
+    int start;
+    int end;
```
Should there be access modifiers for these fields? 🤔 Same with codes in other files.

desmondyst commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,101 @@
+                System.out.println("Nice! I've marked this task as done:\n" +
+                        "[" + tasks.get(counter).getStatusIcon() + "] " + tasks.get(counter).getDescription());
+                storage.saveFile(tasks);
+
```
Perhaps u can remove the newline between `if` and `else if` statements etc? Same go with other lines of code.
```suggestion

```

desmondyst commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,64 @@
+            Files.createDirectories(Paths.get("data"));
+            this.file = new File("data/tasks.txt");
+            file.createNewFile();
+        } catch (IOException e) {
```
Should this catchblock be empty? 💭

desmondyst commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,39 @@
+        return "[" + this.getStatusIcon() + "] " + this.getDescription();
+    }
+
+    String saveStringToFile() {
```
Missing access modifier!

desmondyst commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,17 @@
+
+public class EventTest {
+    @Test
+    public void getTaskTypeTest(){
```
```suggestion
    public void getTaskTypeTest() {
```

desmondyst commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    @Test
+    public void toStringTest(){
```
```suggestion
    public void toStringTest() {
```

desmondyst reviewed (commented):
LGTM, just some small nits I pointed out in the comments. Perhaps u can consider having a `task ` package and putting all the files that `extends Task` inside. Keep up the good work!

cxyterence commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Perhaps you might want to consider a more descriptive variable name like `byDate`, so that readers can easily differentiate between this variable and the other `by` parameter in your constructor

cxyterence commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,28 @@
+
+public class Event extends Task {
+
+    protected LocalDate at;
```
Similar to `Deadline.java`, you could consider a more descriptive variable name like `atDate`

cxyterence commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,24 @@
+package duke;
+public class Todo extends Task {
+
+    protected String by;
```
You can consider removing this variable declaration since it's not being used

cxyterence reviewed (commented):
Your code generally looks good :D 
There are some minute minor details that you might want to take note of

shawnkai committed:
added javadocs

shawnkai committed:
Tweak code to comply with coding standards

shawnkai committed:
edited parser.java and tasklist.java to include the "find" command

shawnkai committed:
Merge branch 'A-JavaDoc'

shawnkai committed:
Merge branch 'branch-A-CodingStandard'

shawnkai committed:
Merge branch 'branch-Level-9'

shawnkai committed:
add dialogbox.java and launcher.java and 2 images for gui, edit duke.java and parser.java

shawnkai committed:
Add assertions to code

Assertions are not existent in the code to verify bugs.

As such, certain bugs might not be detected in code.

Use assert feature to document important assumptions that should hold
at various points in the code.

This is to prevent major bugs in the code such as description of
Tasks being null.

shawnkai committed:
refactor code and improve code quality

shawnkai committed:
Merge pull request #3 from shawnkai/branch-A-Assertions

Add assertions to code

shawnkai committed:
Merge branch 'master' of https://github.com/shawnkai/ip

shawnkai committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Parser.java

shawnkai committed:
Add Reminder Function

Duke is unable to provide a reminder function that can remind the user
of upcoming events or deadlines in the coming week.

Reminder function should be added so that user can easily know what
are the upcoming events and deadlines.

Let's have a reminder function that can allow the user to know about
the upcoming deadlines and events within the next week.

Typing in "Reminder" returns the deadlines and events for the next
7 days. E.g. Typing "Reminder" on 12 September 2022 will return the
deadlines and events from 12 September 2022 (Exclusive) to 19 September
2022 (Exclusive).

shawnkai committed:
Merge branch 'branch-AddAnExtension'

shawnkai committed:
Update Readme and add screenshot of GUI

shawnkai committed:
edit user guide

shawnkai committed:
create new jar file and edit file name from Launcher.java to Main.java

shawnkai committed:
edit ui.png
